tests/js/spec/views/dashboardsV2/widgetBuilder/widgetBuilder.spec.tsx

```diff
@@ -810,6 +815,58 @@ describe('WidgetBuilder', function () {
     expect(handleSave).toHaveBeenCalledTimes(1);
   });
 
+  it('persists the global selection header period when updating a widget', async () => {
+    const widget: Widget = {
+      id: '1',
+      title: 'Errors over time',
+      interval: '5m',
+      displayType: DisplayType.LINE,
+      queries: [
+        {
+          name: 'errors',
+          conditions: 'event.type:error',
+          fields: ['count()', 'count_unique(id)'],
+          aggregates: ['count()', 'count_unique(id)'],
+          columns: [],
+          orderby: '',
+        },
+      ],
+    };
+    const dashboard: DashboardDetails = {
+      id: '1',
+      title: 'Dashboard',
+      createdBy: undefined,
+      dateCreated: '2020-01-01T00:00:00.000Z',
+      widgets: [widget],
+    };
+
+    const {router} = renderTestComponent({
+      dashboard,
+      widget,
+      params: {orgId: 'org-slug', dashboardId: '1'},
+      query: {statsPeriod: '90d'},
+    });
+
+    await screen.findByText('Update Widget');
+    const globalSelectionHeaderTimePeriod = screen.getByTestId(
+      'global-header-timerange-selector'
+    );
+    await within(globalSelectionHeaderTimePeriod).findByText('Last 90 days');
```

Comment:
I would avoid the usage of within. Should we maybe just check if the "Last 90 days" is being displayed on the page?